domain/src/main/kotlin/no/nav/su/se/bakover/domain/søknadsbehandling/Søknadsbehandling.kt

```diff
@@ -1345,47 +1345,40 @@ sealed class KunneIkkeIverksette {
     object KunneIkkeGenerereVedtaksbrev : KunneIkkeIverksette()
 }
 
-private fun List<Søknadsbehandling>.kastHvisDetFinnesEnAnnenÅpenBehandling(behandlingId: UUID) {
-    this
-        .filter { b -> b.id != behandlingId }
-        .filterNot { it.status == BehandlingsStatus.IVERKSATT_INNVILGET || it.status == BehandlingsStatus.IVERKSATT_AVSLAG }
-        .ifNotEmpty {
-            throw IllegalStateException("Det finnes flere behandlinger under behandling")
-        }
-    // TODO jah: Kan på sikt flytte disse valideringene til Sak og/eller database-laget.
-}
-
-private fun List<Søknadsbehandling>.kastHvisDetFinnesEnÅpenBehandling() {
-    this
-        .filterNot { it.status == BehandlingsStatus.IVERKSATT_INNVILGET || it.status == BehandlingsStatus.IVERKSATT_AVSLAG }
-        .ifNotEmpty {
-            throw IllegalStateException("Det finnes flere behandlinger under behandling")
-        }
-    // TODO jah: Kan på sikt flytte disse valideringene til Sak og/eller database-laget.
-}
-
 fun List<Søknadsbehandling>.hentSøknadstypeUtenBehandling(): Søknadstype {
```

Comment:
Kunne kanskje sett for oss en mer "generell" tilnærming hvor en sak hadde muligheten til å finne sitt eget førstegangsvedtak - hvor man deretter brukte denne informasjonen til å avgjøre hvilken "type" som skal velges. Føle som denne fremgangsmåten ville forenklet denne prosessen en del?

Reply:
I dette tilfellet har vi hentet saken, så det gir mening og legge til en wrapper-funksjon i sak som kaller denne funksjonen. Men jeg mener logikken bor sammen med Søknadsbehandling og ikke sak.